Rcb 21 delete private recipe

SocaciVictor opened the pull request.

SocaciVictor committed:
Added the functionality to delete a recipe

I added ENUM for user's role.

I added two buttons. The edit button is created only for layout purposes and has no functionality, while the delete button deletes the chef's recipe.

SocaciVictor committed:
Fixing some bugs

During the creation of the delete button, I noticed some minor issues, such as:

-When updating a recipe, the preparation time was not updated.

-When adding a recipe, the preparation time was not updated in the database, and the kcal/serving was not calculated correctly.

-Also, when updating a recipe, the new quantities of ingredients in the recipe were not updated.

en-catrinel-lupu commented on recipe-box-frontend/src/main/frontend/src/Pages/RecipeViewPage/RecipeViewPage.tsx:
```diff
@@ -25,9 +26,12 @@ const RecipeViewPage: React.FC = () => {
                     <>
                         <RecipeDetails
+                            recipeId={recipeDetails.id}
+                            userId={1}
```
why is the userId hardcoded? it's best if we pass it dynamically

en-catrinel-lupu commented on recipe-box-frontend/src/main/frontend/src/Pages/RecipeViewPage/RecipeViewPage.tsx:
```diff
@@ -25,9 +26,12 @@ const RecipeViewPage: React.FC = () => {
                             name={recipeDetails.name}
                             description={recipeDetails.description}
                             ingredients={recipeDetails.recipeIngredients}
+                            userType={UserType.CHEF}
```
same here, maybe we can pass it dynamically. Check if we can get the role from the api or from localStorage

en-catrinel-lupu commented on recipe-box-frontend/src/main/frontend/src/components/RecipeView/RecipeDetails.tsx:
```diff
@@ -1,18 +1,68 @@
+    const [showDialog, setShowDialog] = useState(false);
+    const navigate = useNavigate();
+
+    const handleDeleteClick = () =>{
```
You could use useCallback on your handlers in order to avoid re-renders. It's better for reusability

en-catrinel-lupu commented on recipe-box-frontend/src/main/frontend/src/components/RecipeView/RecipeDetails.tsx:
```diff
@@ -1,18 +1,68 @@
+            navigate('/recipes/list');
+        } catch (error) {
+            console.error('Error:', error);
+            console.log("Cannot delete Recipe");
```
instead of just logging the error with a text, you could use alert

SocaciVictor committed:
Update after review

I made userId and userType retrieve data from session storage to check whether the current user is logged in or not.

I used useCallback to avoid unnecessary re-renders.

en-marius-gociman commented on recipe-box-frontend/src/main/frontend/src/components/RecipeView/RecipeDetails.tsx:
```diff
@@ -1,28 +1,99 @@
+    }, []);
+
+    const handleDialogClose = useCallback(() => {
+        setShowDialog(false);
```
These can be directly written when you use them, something like
`onClick={()=>setShowDialog(false)}`
in order to avoid moving them to functions, let alone using `useCallback`

en-marius-gociman commented on recipe-box-frontend/src/main/frontend/src/Pages/RecipeViewPage/RecipeViewPage.tsx:
```diff
@@ -17,6 +18,12 @@ const RecipeViewPage: React.FC = () => {
+    const UserId = storedUserId !== null ? parseInt(storedUserId, 10) : undefined;
+
+    const storedRole = sessionStorage.getItem("role");
+    const role = storedRole ? (storedRole as UserType) : undefined;
```
left a comment about these getters in Ionel's PR, these would be better in a Utils/user.ts folder/file
We can clean up later if his change is merged before yours

en-alexandra-costache commented on recipe-box-backend/src/main/java/com/endava/recipebox/service/impl/RecipeServiceImpl.java:
```diff
@@ -181,12 +200,39 @@ public String updateRecipe(RecipeEditRequestDTO recipeAddRequestDTO, Long userId
+            List<RecipeIngredient> updatedIngredients = recipeAddRequestDTO.getIngredients().stream()
+                    .map(ingredientDTO -> {
+                        Ingredient ingredient = ingredientRepository.findById(ingredientDTO.getIngredientId())
+                                .orElseThrow(() -> new EntityNotFoundException("No ingredient found in the DB."));
```
I would change the message to see the id of the ingredient that was not found

SocaciVictor committed:
Update the message

SocaciVictor committed:
Merge remote-tracking branch 'origin/main' into RCB-21-Delete-Private-Recipe

# Conflicts:
#	recipe-box-backend/src/main/java/com/endava/recipebox/mapper/RecipeMapper.java
#	recipe-box-frontend/src/main/frontend/src/enums/User.ts

SocaciVictor committed:
Merge remote-tracking branch 'origin/main' into RCB-21-Delete-Private-Recipe

SocaciVictor committed:
Update callback functions

SocaciVictor committed:
Update RecipeLoader.ts

SocaciVictor committed:
Moved style in RecipeDetails.css
